### PR TITLE
Updated CNV analysis 

### PR DESCRIPTION
**Pull request by josephineyates**
This PR does two main things: 
- adds the number of patients showing either a gain or a loss at this specific locus
- adds the possibility to call CNVs from a subclonal perspective

### CODE REVIEW

File: cansig/cnvanalysis/differentialcnvs.py [hunk 1]
```diff
@@ -27,13 +29,54 @@ def discretize_cnv(data: anndata.AnnData, cnv_key: str = "X_cnv") -> pd.DataFram
...
+
+    pos_cnv_array = cnv_array[cnv_array >= 0].fillna(0)
+    neg_cnv_array = cnv_array[cnv_array <= 0].fillna(0)
+
+    sub_pos_cnv_array = pd.concat([pos_cnv_array, data.obs[subclonal_key]], axis=1)
+    sub_neg_cnv_array = pd.concat([neg_cnv_array, data.obs[subclonal_key]], axis=1)
+
+    mean_sub_pos = sub_pos_cnv_array.groupby(subclonal_key).mean()
+    mean_sub_pos[mean_sub_pos > SUBCLONAL_MAJORITY] = 1
+    mean_sub_pos = mean_sub_pos[mean_sub_pos > SUBCLONAL_MAJORITY].fillna(0)
```
Comment (FlorianBarkmann) [inline cansig/cnvanalysis/differentialcnvs.py:44]: 30% seems low to me.

File: cansig/cnvanalysis/differentialcnvs.py [hunk 2]
```diff
@@ -42,26 +85,50 @@ def get_diff_cnv(
...
+                rloss = np.nan
+            else:
+                pval = diff_function(cl_cnv[col].values, rest_cnv[col].values)[1]
+                pgain = (cl_cnv[col] > 0).sum() / cl_cnv.shape[0]
+                ploss = (cl_cnv[col] < 0).sum() / cl_cnv.shape[0]
+                rgain = (rest_cnv[col] > 0).sum() / rest_cnv.shape[0]
+                rloss = (rest_cnv[col] < 0).sum() / rest_cnv.shape[0]
+
+            # get the p value of the test
+            all_results[str(cluster) + "_pvalues"].append(pval)
```
Comment (FlorianBarkmann) [inline cansig/cnvanalysis/differentialcnvs.py:115]: I think this could by done cleaner by initializing the dataframe from the beginning and just filling it instead of first building dicts and converting them to a dataframe.

**Review by FlorianBarkmann — APPROVED** [review]
LGTM

File: cansig/cnvanalysis/differentialcnvs.py (hunk 2 above)
Comment (josephineyates) [reply cansig/cnvanalysis/differentialcnvs.py:115]: Tbh I think both versions are as clear and it just makes it very readable imo

File: cansig/cnvanalysis/differentialcnvs.py (hunk 1 above)
Comment (josephineyates) [reply cansig/cnvanalysis/differentialcnvs.py:44]: ahaha fair, I randomly picked but I can change to 0.5
